search/docker-compose-dev.yml

```diff
@@ -0,0 +1,121 @@
+version: "3"
```

Comment:
I would rather have a single file / template than maintaining two parallel .ymls. I'm okay with it for now, but inevitably copied code/config gets out of sync and is a pain to maintain.

Reply:
So you prefer having one file with multiple parameter in the docker command, like `reactive_search_dockerfile=Dockerfile-dev ... docker-compose up`? Also in that way you need to comment volume every time switching, which I didn't find a solution to resolve and it's much more painful than just switching between different yml files.

Reply:
Look, I'm okay with it as it is, but in general I dislike having to maintain two parallel different implementations. Something like https://github.com/sinzlab/jinja-compose solves this problem.

Reply:
The documentation is not clear to me, like where the param N_GPU is defined as below. It seems it needs to generate yml file before running docker-compose. The IF condition block might be able to include or exclude volume, but I reckon we keep it as what it is now till we have major changes.
```
version: "2"
services:
  {% for i in range(N_GPU) %}
  notebook{{i}}:
    image: sinzlab/pytorch
    ports:
      - "300{{i}}:8888"
    environment:
      - NVIDIA_VISIBLE_DEVICES={{i}}
    volumes:
      - ./notebooks:/notebooks
  {% endfor %}
```

Reply:
Leave it alone for now.

Reply:
Yes, it seems that I was misled by a stackoverflow comment to think this was a mature and usable piece of software. I would've thought that it either had a way to insert config, or would provide env to jinja.

maybe the docker-compose community disagrees with me and prefers to maintain multiple copies, or are using other approaches to template their docker-compose files?

Reply:
https://pypi.org/project/Docker-Compose-Templer/ seems a more mature cousin, but still not quite feature complete. An OSS project? :D